From 2227c95be6267f7ad1ec3ad2fc09dbb5146bae0f Mon Sep 17 00:00:00 2001
From: Jordi Mallach <jordi@debian.org>
Date: Wed, 15 May 2013 19:11:47 +0200
Subject: [PATCH] Check for and include libavutil.h to avoid unresolved
 symbols.

The a52 plugin uses av_free(), but does not check for libavutil
availability. Patch configure.in to check for the module, and include
libavutil.h.

Signed-off-by: Jordi Mallach <jordi@debian.org>
Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 a52/pcm_a52.c | 1 +
 configure.in  | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/a52/pcm_a52.c b/a52/pcm_a52.c
index c40aadc..75c68bd 100644
--- a/a52/pcm_a52.c
+++ b/a52/pcm_a52.c
@@ -26,6 +26,7 @@
 #include <alsa/pcm_external.h>
 #include <alsa/pcm_plugin.h>
 #include AVCODEC_HEADER
+#include <libavutil/avutil.h>
 
 #if LIBAVCODEC_VERSION_MAJOR >= 53 && LIBAVCODEC_VERSION_MINOR >= 34
 #include <libavutil/audioconvert.h>
diff --git a/configure.in b/configure.in
index 6f345f5..bf8676b 100644
--- a/configure.in
+++ b/configure.in
@@ -67,7 +67,7 @@ AC_ARG_ENABLE([avcodec],
       AS_HELP_STRING([--disable-avcodec], [Don't build plugins depending on avcodec (a52)]))
 
 if test "x$enable_avcodec" != "xno"; then
-  PKG_CHECK_MODULES(AVCODEC, [libavcodec], [HAVE_AVCODEC=yes], [HAVE_AVCODEC=no])
+  PKG_CHECK_MODULES(AVCODEC, [libavcodec libavutil], [HAVE_AVCODEC=yes], [HAVE_AVCODEC=no])
 fi
 
 if test "x$HAVE_AVCODEC" = "xno"; then
-- 
1.7.11.7

